sdk/testutil/server.go

```diff
@@ -333,51 +340,46 @@ func (s *TestServer) Stop() error {
 	return s.cmd.Wait()
 }
 
-type failer struct {
-	failed bool
-}
-
-func (f *failer) Log(args ...interface{}) { fmt.Println(args...) }
-func (f *failer) FailNow()                { f.failed = true }
-
 // waitForAPI waits for only the agent HTTP endpoint to start
 // responding. This is an indication that the agent has started,
 // but will likely return before a leader is elected.
 func (s *TestServer) waitForAPI() error {
-	f := &failer{}
-	retry.Run(f, func(r *retry.R) {
+	var failed bool
+	timer := retry.TwoSeconds()
+
+	deadline := time.Now().Add(timer.Timeout)
+	for !time.Now().After(deadline) {
+		time.Sleep(timer.Wait)
+
 		resp, err := s.HTTPClient.Get(s.url("/v1/agent/self"))
 		if err != nil {
-			r.Fatal(err)
+			failed = true
+			continue
 		}
-		defer resp.Body.Close()
-		if err := s.requireOK(resp); err != nil {
-			r.Fatal("failed OK response", err)
+		resp.Body.Close()
+
+		if err = s.requireOK(resp); err != nil {
+			failed = true
+			continue
 		}
-	})
-	if f.failed {
-		return errors.New("failed waiting for API")
+		failed = false
+	}
+	if failed {
+		return fmt.Errorf("api unavailable")
```

Comment:
This code looks good but maybe would benefit from a comment explaining why we aren't just using `retry` here? Otherwise it might get "improved" later to re-use `retry`.

Reply:
Good call, just added a comment. Thanks!